### PR TITLE
Move some secrets outside the repository

### PR DESCRIPTION
**Pull request by mokagio**
Moves the low-hanging fruits in terms of secrets outside of the repository:

- Fastlane's `google-upload-cerdentials.json`
- The debug keystore

## To test

Checkout this branch and run `bundle exec fastlane run configure_apply`, then `ls ~/.configure/wordpress-android/secrets`, you should see:

```
➜ ls ~/.configure/wordpress-android/secrets

debug_a8c.keystore             google-upload-credentials.json
```

**Individual tests**

`debug_a8c.keystore`: As far as I can see, the debug keystore is only used in CI, in particular in CircleCI? I run all the CircleCI steps and they worked. The fact that we don't need to worry about that keystore for local development seem to be confirmed by the [_Sign your debug build_](https://developer.android.com/studio/publish/app-signing#debug-mode) documentation.

`google-upload-credentials.json`: This file is used by the `upload_to_play_store` Fastlane action. I checked the action [documentation](https://docs.fastlane.tools/actions/upload_to_play_store/) and couldn't find a way to perform a dry-run. I saw there is a `validate_only` option, but the documentation says "Only validate changes with Google Play rather than actually publish," which doesn't make it clear to me whether data would be uploaded or not. I'm not familiar with the upload steps in this repo and would like to avoid uploading stuff accidentally.

@AliSoftware and @oguzkocer, I think you are the most knowledgeable when it comes to this part of the WordPress Android codebase: any suggestion on how to further verify these changes before merging? Thanks!

## Regression Notes
1. Potential unintended areas of impact

I don't think these changes may impact areas other than what specified above, but I'll defer to those who have more know-how in Android and this repo.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
N.A.

3. What automated tests I added (or what prevented me from doing so)
N.A.

---

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes. **N.A.**
- [x] I have considered adding accessibility improvements for my changes. **N.A.**
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary. **N.A.**

### CODE REVIEW

File: .configure
Comment (mokagio) [inline .configure:28]: This was technically already outside the repository, but I moved it to the new default location: `~/.configure/<project-name>/secrets/`

**Comment by oguzkocer** [comment]
> @AliSoftware and @oguzkocer, I think you are the most knowledgeable when it comes to this part of the WordPress Android codebase: any suggestion on how to further verify these changes before merging? Thanks!

The code changes look good to me (not tested it yet), but I feel @jkmassel and @loremattei would be the best candidates to review this because all these were setup even before my rotation and I've never had to make any changes to it. (so still don't know much about them) I'd just feel much better with their 👍 @AliSoftware might have the necessary experience though, so I am just speaking for myself :)

File: .configure
Comment (AliSoftware) [reply .configure:28]: Thanks I appreciate the standardization 👍

I'm not familiar with how/where this keystore file is used in the codebase though, is it expected that there was nothing changed in the rest of the repo (like gradle file or properties file or project settings) to update to that new path?

**Review by AliSoftware — COMMENTED** [review]
Reviewing from my phone so I wasn't yet able to test, but the diff makes sense to me.

Left a question about the `debug_a8c.keystore` (haven't found any info about it in the README and can't MGS from phone so not sure how it's used) but apart from that the rest looks good. (I'll leave for someone able to test `configureApply` on a Mac to approve)

**Comment by mokagio** [comment]
Explicitly asked @loremattei and @jkmassel review via GitHub as suggested by @oguzkocer.

@oguzkocer @AliSoftware thanks for taking a look at this nevertheless 🙇‍♂️

File: .configure
Comment (loremattei) [reply .configure:28]: Nice! But as @AliSoftware suggested, you have to update the configuration in `gradle.properties` to match this new path in order to make the debug certificate work on local builds.

**Review by loremattei — DISMISSED** [review]

**Comment by loremattei** [comment]
This looks good to me 👍 Nice work @mokagio! Thanks for tackling it. 
I've only found one missing change to the configuration that we need to make. Other than that, I think it's ok.

**Comment by mokagio** [comment]
> you have to update the configuration in `gradle.properties` to match this new path in order to make the debug certificate work on local builds.

😮 Thanks @loremattei @AliSoftware! I used `git grep` to look for usages of `debug_a8c.keystore` but `gradle.properties` is ignored.

**Comment by mokagio** [comment]
@loremattei thanks again for the review and the explanation. I have addressed your comment. Mind having another look? Thanks!

**Review by jkmassel — APPROVED** [review]
This makes sense to me, and I don't see any immediate issues with it. I also see that `gradle.properties.enc` is now changing, so you should be good to go

**Comment by mokagio** [comment]
Thanks @jkmassel.

I'll leave it up to @oguzkocer as the release engineer currently on this app to decide whether to merge it and try it out for 17.9, or merge it after the code freeze to avoid possible issues delaying it.

Sorry Oguz for not following up on this earlier. It might have avoided this dilemma.

**Comment by oguzkocer** [comment]
@mokagio WPAndroid release manager is @AliSoftware - I am just covering for him this week :) So, I am leaving it to him I guess 😅

**Comment by AliSoftware** [comment]
Thanks @mokagio 👍 

Missed this PR before code-freeze, so I'll likely only test it during next sprint instead of this one, will report if I see any issue 👍  (and I'll try to remember to `git clean` by then to remove secrets files from old location…)

**Pull request merged**
